src/expycted/matchers/equal_matcher.py

```diff
@@ -11,4 +12,13 @@ class EqualMatcher(BaseMatcher):
     OPERATION = "=="
 
     def _matches(self, *, expected: Any) -> bool:
-        return self._expectation.actual == expected
+        if not self._expectation.qualifiers.weak:
+            return self._expectation.actual == expected
+
+        check_callbacks = [
+            lambda: str(self._expectation.actual) == str(expected),
+            lambda: pickle.dumps(self._expectation.actual) == pickle.dumps(expected),
+            lambda: self._expectation.actual == expected,
+        ]
```

Comment:
Maybe it would be worth it to move these to *class methods* so that they don't need to be redefined everytime `_matches` is called.